conseil-lorre/src/main/scala/tech/cryptonomic/conseil/indexer/ethereum/EthereumOperations.scala

```diff
@@ -0,0 +1,115 @@
+package tech.cryptonomic.conseil.indexer.ethereum
+
+import cats.effect.{Concurrent, Resource}
+import com.typesafe.scalalogging.LazyLogging
+import fs2.Stream
+import slickeffect.Transactor
+
+import tech.cryptonomic.conseil.common.config.Platforms.EthereumBatchFetchConfiguration
+import tech.cryptonomic.conseil.common.rpc.RpcClient
+import tech.cryptonomic.conseil.common.ethereum.EthereumPersistence
+import tech.cryptonomic.conseil.common.ethereum.rpc.EthereumClient
+import tech.cryptonomic.conseil.indexer.config.{Custom, Depth, Everything, Newest}
+
+/**
+  * Ethereum operations for Lorre.
+  *
+  * @param ethereumClient Ethereum client instance
+  * @param persistence Ethereum persistence instance
+  * @param tx [[slickeffect.Transactor]] to perform a Slick operations on the database
+  * @param batchConf Batch processing configuration
+  */
+class EthereumOperations[F[_]: Concurrent](
+    ethereumClient: EthereumClient[F],
+    persistence: EthereumPersistence[F],
+    tx: Transactor[F],
+    batchConf: EthereumBatchFetchConfiguration
+) extends LazyLogging {
+
+  /**
+    * Start Lorre with mode defined with [[Depth]].
+    *
+    * @param depth Can be: Newest, Everything or Custom
+    */
+  def loadBlocksAndLogs(depth: Depth): Stream[F, Unit] =
+    Stream
+      .eval(tx.transact(persistence.getLatestIndexedBlock))
+      .zip(ethereumClient.getMostRecentBlockNumber.map(Integer.decode))
+      .flatMap {
+        case (block, mostRecentBlockNumber) =>
+          val range = depth match {
+            case Newest => block.map(_.number + 1).getOrElse(1) to mostRecentBlockNumber
+            case Everything => 1 to mostRecentBlockNumber
+            case Custom(depth) => (mostRecentBlockNumber - depth) to mostRecentBlockNumber
```

Comment:
Might be easy to check if we're asking too deep a level, here.

- use 1 if depth > mostRecent
- use block.number if higher than mostRecent - depth

Reply:
I'm not sure what you did cover the cases I had in mind, that is:
 1. passed depth is higher than the mostRecent height on the chain: start from 1 to mostRecent
 2. same as before, but instead of starting from the first block, we might want to behave as if Newest was passed, so we start from latestIndexedBlock number + 1, if available